turbopack/crates/turbopack-ecmascript/src/references/esm/url.rs

```diff
@@ -47,37 +47,37 @@ pub enum UrlRewriteBehavior {
 /// referenced file to be imported/fetched/etc.
 #[turbo_tasks::value]
 pub struct UrlAssetReference {
-    origin: Vc<Box<dyn ResolveOrigin>>,
-    request: Vc<Request>,
-    rendering: Vc<Rendering>,
-    ast_path: Vc<AstPath>,
-    issue_source: Vc<IssueSource>,
+    origin: ResolvedVc<Box<dyn ResolveOrigin>>,
+    request: ResolvedVc<Request>,
+    rendering: ResolvedVc<Rendering>,
+    ast_path: ResolvedVc<AstPath>,
+    issue_source: ResolvedVc<IssueSource>,
     in_try: bool,
-    url_rewrite_behavior: Vc<UrlRewriteBehavior>,
+    url_rewrite_behavior: ResolvedVc<UrlRewriteBehavior>,
 }
 
 #[turbo_tasks::value_impl]
 impl UrlAssetReference {
     #[turbo_tasks::function]
-    pub fn new(
+    pub async fn new(
         origin: Vc<Box<dyn ResolveOrigin>>,
         request: Vc<Request>,
         rendering: Vc<Rendering>,
         ast_path: Vc<AstPath>,
         issue_source: Vc<IssueSource>,
         in_try: bool,
         url_rewrite_behavior: Vc<UrlRewriteBehavior>,
-    ) -> Vc<Self> {
-        UrlAssetReference {
-            origin,
-            request,
-            rendering,
-            ast_path,
-            issue_source,
+    ) -> Result<Vc<Self>> {
+        Ok(UrlAssetReference {
+            origin: origin.to_resolved().await?,
+            request: request.to_resolved().await?,
+            rendering: rendering.to_resolved().await?,
+            ast_path: ast_path.to_resolved().await?,
+            issue_source: issue_source.to_resolved().await?,
             in_try,
-            url_rewrite_behavior,
+            url_rewrite_behavior: url_rewrite_behavior.to_resolved().await?,
         }
-        .cell()
+        .cell())
```

Comment:
You can use `ResolvedVc` in the arguments instead to not have to call `to_resolved()` on everything